skimage/metrics/_variation_of_information.py

```diff
@@ -104,8 +104,8 @@ def _vi_tables(im_true, im_test, table=None, ignore_labels=()):
 
     # use sparse matrix linear algebra to compute VI
     # first, compute the inverse diagonal matrices
-    px_inv = sparse.diags(_invert_nonzero(px))
-    py_inv = sparse.diags(_invert_nonzero(py))
+    px_inv = sparse.dia_array((_invert_nonzero(px), 0), shape=(n := len(px), n))
+    py_inv = sparse.dia_array((_invert_nonzero(py), 0), shape=(n := len(py), n))
```

Comment:
Woah, I had to do a second take to understand what's going on here. Maybe we can simplify this somewhat with simpler Python paradigmens
```suggestion
    px_inv = sparse.dia_array((_invert_nonzero(px), 0), shape=(px.size,) * 2)
    py_inv = sparse.dia_array((_invert_nonzero(py), 0), shape=(py.size,) * 2)
```

What do you think?

Reply:
Or even simpler using just `(px.size, px.size)`..